### PR TITLE
Get search from URL

### PR DESCRIPTION
**Pull request by ludeeus**
Use the value from `?domain=[domain]#integrations` to initialize the filter on the integrations tab if present 

https://deploy-preview-294--home-assistant-analytics.netlify.app/?domain=esphome#integrations

if the requested integration is hidden by default (default/internal), the `_showDefaultAndInternal` property is set to `true` 
https://deploy-preview-294--home-assistant-analytics.netlify.app/?domain=analytics#integrations

### CODE REVIEW

File: site/src/analytics-element.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export class AnalyticsElement extends LitElement {
 
   private _pageChanged() {
     this._currentPage =
-      window.location.hash.replace("#", "") || "installations";
+      window.location.hash.replace("#", "").split("?")[0] || "installations";
```
Comment (bramkragten) [inline site/src/analytics-element.ts:149]: Shouldn't you just put the hash after the search params?

File: site/src/analytics-integrations.ts [hunk 2]
```diff
@@ -205,6 +218,7 @@ export class AnalyticsIntegrations extends LitElement {
   private _clearFilter() {
     this._currentTablePage = 0;
     this._filter = "";
+    window.location.search = "";
```
Comment (balloob) [inline site/src/analytics-integrations.ts:221]: You should do this after you consumed the filter from the URL. Not every time.

File: site/src/analytics-integrations.ts [hunk 3]
```diff
@@ -52,7 +52,17 @@ export class AnalyticsIntegrations extends LitElement {
...
     super.firstUpdated(_changedProperties);
 
-    this.getData();
+    const query = new URLSearchParams(window.location.search);
+    this._filter = query.get("search")?.toLowerCase() || "";
+    this.getData().then(() => {
+      if (
+        this._filter !== "" &&
+        (DEFAULT_DOMAINS.includes(this._filter) ||
+          this._integrationDetails[this._filter].quality_scale === "internal")
```
Comment (balloob) [inline site/src/analytics-integrations.ts:61]: This is a bit of a weird check because it assumes it's an exact match which is not what "filter" implies ?

Comment (ludeeus) [reply site/src/analytics-integrations.ts:61]: This is only on init, so it only applies when using query param

File: site/src/analytics-integrations.ts (hunk 2 above)
Comment (ludeeus) [reply site/src/analytics-integrations.ts:221]: I'll move all handling to the main element

File: site/src/analytics-integrations.ts (hunk 3 above)
Comment (ludeeus) [reply site/src/analytics-integrations.ts:63]: Changed it to "domain" to make it clearer

File: site/src/analytics-integrations.ts (hunk 2 above)
Comment (ludeeus) [reply site/src/analytics-integrations.ts]: Added a guard

File: site/src/analytics-header.ts [hunk 4]
```diff
@@ -11,23 +11,19 @@ export class AnalyticsHeader extends LitElement {
...
-            html`<div
-              .page=${page}
-              class="page"
-              @click=${this._pageSelected}
-              ?selected=${page === this.currentPage}
-            >
-              ${page}
-            </div>`
+            html`<a href="/#${page}">
+              <div
```
Comment (balloob) [inline site/src/analytics-header.ts:15]: why even have a div

**Review by balloob — APPROVED** [review]

File: site/src/analytics-header.ts (hunk 4 above)
Comment (ludeeus) [reply site/src/analytics-header.ts:15]: Good question...

Comment (ludeeus) [reply site/src/analytics-header.ts]: removed :+1:

**Pull request merged**